nixos/modules/services/web-apps/cryptpad.nix

```diff
@@ -0,0 +1,213 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.cryptpad;
+
+  inherit (lib) strings mdDoc mkOption types mkIf mkMerge;
+
+  # The Cryptpad configuration file isn't JSON, but a JavaScript source file that assigns a JSON value
+  # to a variable.
+  cryptpadConfigFile = builtins.toFile "cryptpad_config.js" ''
+    module.exports = ${builtins.toJSON cfg.config}
+  '';
+
+  # Derive domain names for Nginx configuration from Cryptpad configuration
+  mainDomain = strings.removePrefix "https://" cfg.config.httpUnsafeOrigin;
+  sandboxDomain = if isNull cfg.config.httpSafeOrigin then mainDomain else strings.removePrefix "https://" cfg.config.httpSafeOrigin;
+
+in {
+  options.services.cryptpad = {
+    enable = lib.mkEnableOption "cryptpad";
+
+    package = lib.mkPackageOption pkgs "cryptpad" {};
+
+    configureNginx = mkOption {
+      description = mdDoc ''
+        Configure Nginx as a reverse proxy for Cryptpad.
+        Note that this makes some assumptions on your setup, and sets settings that will
+        affect other virtualHosts running on your Nginx instance, if any.
+        Alternatively you can configure a reverse-proxy of your choice.
+      '';
+      type = types.bool;
+      default = false;
+    };
+
+    confinement = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        FIXME: Enables 'confinement' (explain what it does?)
+        Enable the nixos systemd.services.cryptpad.confinement setting, including the necessary
+        extra bind mounts.
+      '';
+    };
+
+    config = mkOption {
```

Comment:
Maybe this should be renamed from `config` to `settings` in line with [RFC0042](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md#part-1-structural-settings-instead-of-stringly-extraconfig)?

Reply:
Fixed.